app/services/coin_market_cap_service.rb

```diff
@@ -1,40 +1,168 @@
 require_relative '../../lib/tasks/batch_process'
 
 class CoinMarketCapService
```

Comment:
Totally my bad, but basically this file is historically named incorrectly; it should be called `NorthPoleProxyService` or something since it only retrieves information from the NorthPole CMC proxy, not CMC directly. There's no need to rename it though since we're planning on deleting this file soon anyways.

Instead, can we create a new `CoinMarketCapProService` file that actually contains only logic pertaining to the new CMC Pro API and revert this file back to the original?

This will also make it easier to perform the code review as well as delete this file when it's no longer needed. Thanks!

Reply:
sure